lib/auth/auth_with_roles.go

```diff
@@ -6318,6 +6318,13 @@ func (a *ServerWithRoles) CreateAuthenticateChallenge(ctx context.Context, req *
 
 // CreatePrivilegeToken is implemented by AuthService.CreatePrivilegeToken.
 func (a *ServerWithRoles) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (*types.UserTokenV3, error) {
+	// Device trust: authorize device before issuing a privileged token without an MFA response.
```

Comment:
`CreatePrivilegeToken` is used to create a token to then create a register challenge without the normal MFA check, which includes the check to see if the user already has no MFA device yet. I could refactor it a bit to only impact `CreateRegisterChallenge`, let me see if it's not too much work.

Reply:
So as it turns out, we do need to do the device trust check on `CreatePrivilegeToken`. This endpoint uses the user's credentials to generate a privileged Token either. An MFA check is required if the user has a registered device, but otherwise a `privilege_exception` token is created without the MFA check.

The Proxy then uses the user-created token to create a register challenge on behalf of the user. Enforcing device trust at this step won't work, so instead we do it when the `privilege_exception` token is created.

I had attempted some refactors to consolidate logic, but it all fell through once I realized this limitation.

Reply:
Could we leave a comment here saying that we do this check because of privilege tokens? (Ie, a summary of what you just explained?)